src/libcore/macros.rs

```diff
@@ -644,181 +644,720 @@ macro_rules! uninitialized_array {
     }
 
     /// Expands to the line number on which it was invoked.
     ///
-    /// For more information, see the documentation for [`std::line!`].
+    /// With [`column!`] and [`file!`], these macros provide debugging information for
+    /// developers about the location within the source.
+    ///
+    /// The expanded expression has type `u32` and is 1-based, so the first line
+    /// in each file evaluates to 1, the second to 2, etc. This is consistent
+    /// with error messages by common compilers or popular editors.
+    /// The returned line is *not necessarily* the line of the `line!` invocation itself,
+    /// but rather the first macro invocation leading up to the invocation
+    /// of the `line!` macro.
+    ///
+    /// [`column!`]: macro.column.html
+    /// [`file!`]: macro.file.html
+    ///
+    /// # Examples
     ///
-    /// [`std::line!`]: ../std/macro.line.html
+    /// ```
+    /// let current_line = line!();
+    /// println!("defined on line: {}", current_line);
+    /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    #[rustc_doc_only_macro]
-    macro_rules! line { () => ({ /* compiler built-in */ }) }
+    #[rustc_builtin_macro] #[rustc_macro_transparency = "semitransparent"]
+    pub macro line { () => ({ /* compiler built-in */ }) }
 
-    /// Expands to the column number on which it was invoked.
+    /// Expands to the column number at which it was invoked.
     ///
-    /// For more information, see the documentation for [`std::column!`].
+    /// With [`line!`] and [`file!`], these macros provide debugging information for
+    /// developers about the location within the source.
     ///
-    /// [`std::column!`]: ../std/macro.column.html
+    /// The expanded expression has type `u32` and is 1-based, so the first column
+    /// in each line evaluates to 1, the second to 2, etc. This is consistent
+    /// with error messages by common compilers or popular editors.
+    /// The returned column is *not necessarily* the line of the `column!` invocation itself,
+    /// but rather the first macro invocation leading up to the invocation
+    /// of the `column!` macro.
+    ///
+    /// [`line!`]: macro.line.html
+    /// [`file!`]: macro.file.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let current_col = column!();
+    /// println!("defined on column: {}", current_col);
+    /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    #[rustc_doc_only_macro]
-    macro_rules! column { () => ({ /* compiler built-in */ }) }
+    #[rustc_builtin_macro] #[rustc_macro_transparency = "semitransparent"]
+    pub macro column { () => ({ /* compiler built-in */ }) }
 
-    /// Expands to the file name from which it was invoked.
+    /// Same as `column`, but less likely to be shadowed.
+    #[unstable(feature = "__rust_unstable_column", issue = "0",
+               reason = "internal implementation detail of the `column` macro")]
```

Comment:
It's not an implementation detail of the column macro, but of the `panic!` macro. `column!()` is builtin itself, as well. See #42938, #43057, #43735.

Reply:
Also, possibly it might not be needed in the future, as pointed out by @matklad: https://github.com/rust-lang/rust/issues/48781#issuecomment-493427139

Reply:
This PR is exactly the prerequisite for `$crate::column!` working.
Right now it doesn't work because `column` is a built-in macro rather than a standard library macro.

Once this PR lands, built-in macros will be available through stable paths in the standard library, and I'll submit the next PR hygienizing their use in the standard libary (similarly to https://github.com/rust-lang/rust/pull/61629) and removing `__rust_unstable_colum`.

Reply:
@petrochenkov you should still change the message though.

Reply:
Yeah, will do.